workspaces/diff-engine/src/projections/hash.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{RfcEvent, SpecEvent};
+use cqrs_core::{Aggregate, AggregateEvent, Event};
+use serde_hashkey::to_key;
+use sha2::{Digest, Sha256};
+use std::{collections::BTreeMap, hash::Hash};
+
+#[derive(Debug, Clone)]
+pub struct SpecHashProjection {
+  hasher: Sha256,
+}
+
+impl SpecHashProjection {
+  pub fn to_string(&self) -> String {
+    format!("{:x}", self.hasher.clone().finalize())
+  }
+
+  pub fn with_event(&mut self, evt: SpecEvent) {
+    let key = to_key(&evt).unwrap();
```

Comment:
It's not _too_ different but if we can get away with sticking with serde_json and the answer here: https://stackoverflow.com/questions/42723065/how-to-sort-hashmap-keys-when-serializing-with-serde
then we can also use that to make our snapshots deterministic everywhere we use hashmaps, which is valuable

Reply:
also not having the extra dependency